### PR TITLE
[BUG-FIX] Large Files now encrypt properly

### PR DESCRIPTION
**Pull request by efabens**
- Fix to encryptFile method in PlainCrypto
- Test to ensure that large files encrypt and decrypt successfully

### CODE REVIEW

**Review by fernviridian — APPROVED** [review]
Nice catch!

**Review by galxy25 — APPROVED** [review]

File: e3db-crypto-android/src/main/java/com/tozny/e3db/AndroidCrypto.java [hunk 1]
```diff
@@ -212,7 +212,7 @@ public File encryptFile(File file, byte[] secretKey) throws IOException, E3DBEnc
 
       for(int headAmt = source.read(head), nextAmt = source.read(next);
           headAmt != -1;
-          headAmt = nextAmt, head = next, nextAmt = source.read(next)) {
+          headAmt = nextAmt, head = next.clone(), nextAmt = source.read(next)) {
```
Comment (lkwdwrd) [inline e3db-crypto-android/src/main/java/com/tozny/e3db/AndroidCrypto.java:215]: This is kind of nuts -- and also why I actually really really like immutability as a concept.

**Review by lkwdwrd — APPROVED** [review]
This looks great. I'm sorry you had to go spelunking like this to find it, but super happy with the result.

**Review by lkwdwrd — APPROVED** [review]

File: e3db-crypto-android/src/main/java/com/tozny/e3db/AndroidCrypto.java (hunk 1 above)
Comment (efabens) [reply e3db-crypto-android/src/main/java/com/tozny/e3db/AndroidCrypto.java:215]: yup, the random byte pattern in java is odd

**Pull request merged**